sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelation.scala

```diff
@@ -58,18 +57,29 @@ import org.apache.spark.util.Utils
  *      thrown during job commitment, also aborts the job.
  */
 private[sql] case class InsertIntoHadoopFsRelation(
-    @transient relation: HadoopFsRelation,
+    outputPath: Path,
+    partitionColumns: Seq[Attribute],
+    bucketSpec: Option[BucketSpec],
+    fileFormat: FileFormat,
```

Comment:
It should probably be a class for long term compatibility reasons, but I would defer that to the JIRA about finalizing this interface.